javatests/arcs/showcase/typevariables/VariableEntityTest.kt

```diff
@@ -8,7 +8,7 @@ import org.junit.runner.RunWith
 import org.junit.runners.JUnit4
 
 @RunWith(JUnit4::class)
-class GenerationBehaviorTest {
+class VariableEntityTest {
```

Comment:
The showcase is not meant to be unit tests - it's meant to be a realistic, convincing example of usage in the scenario that reminds of a use cases that the feature was built for.

This tests that you are now moving here are not that, they are more list unit tests and would be noise to the potential customer of the feature IMHO.

For type variables we should start with something like this:

```
particle Producer
  data: writes Product {sku: Text, name: Text, price: Number}

particle SkuRedactor
  input: reads [~a with {sku: Text}]
  output: writes [~a]

particle Consumer
  data: reads Product {sku: Text, name: Text, price: Number}

recipe
  beforeRedaction: create
  afterRedaction: create
  Producer
    data: beforeRedaction
  SkuRedactor
    input: beforeRedaction
    output: afterRedaction
  Consumer
    data: afterRedaction
```

The showcase test should contain:

- The above recipe
- Implementation of particles
- The code that starts an arc, and asserts that the Producer produced say 3 products, SkuRedactor did some transformation of SKUs, and the consumer saw products as they were produced but with SKUs transformed.

At a later stage we should also add a test for the `~a with {*}` scenario, but first lets implement that.

Reply:
If we're going to have a redactor example, shouldn't there be some claims on the output thats redacted?